internal/web_server/web/handlers/api/admin_api.go

```diff
@@ -310,25 +318,38 @@ func (aah *AdminApiHandler) createQuestion(c echo.Context) error {
 		return echo.NewHTTPError(http.StatusBadRequest, errorText)
 	}
 
-	// Check if the article URL is in the database
-	article, err := articles.GetArticleByURL(aah.sharedData.DB, articleURL)
-	if err != nil && err != sql.ErrNoRows {
-		return echo.NewHTTPError(http.StatusBadRequest, "Failed to get article ID")
-	}
+	article := &articles.Article{}
 
-	// If not in DB, fetch the relevant article data and add it to the DB
-	if article == nil {
-		tempArticle, err := articles.GetSmpArticleByURL(articleURLString)
-		if err != nil {
-			return echo.NewHTTPError(http.StatusBadRequest, "Failed to fetch article data")
+	// Only add article to DB if it is not empty.
+	// I.e. allow for no article, but not invalid article.
+	if articleURLString != "" {
+		// Check if the article URL is in the database
+		article, err = articles.GetArticleByURL(aah.sharedData.DB, articleURL)
+		if err != nil && err != sql.ErrNoRows {
```

Comment:
if thereis an error OTHER than `sql.ErrNoRows`, you return status bad request? Is everything that can go wrong other than no rows the clients fault? i believe there might be a 500 (internal server error) if something goes sideways on the server side as well.
To be frank i dont know exactly what  this function does so i might be wrong here. But i think you can have better errors anyway. If matching for possible sql errors or other faults is tedious here, consider defining own error types in model and checking for those

Reply:
on second look, dont you mean that error no rows found should do bad request (or not found), and any other should turn into 500 (ie just return in)? if so you'll need nested if, first to ckeck if err != nil, then in the body check if the error is sql.NoRows -> 400/404/whatever, and else return err

Reply:
If there are no rows found, it means the article does not exist in the database already. That is allowed. We will then simply add it to the database. That is why I check for any error beside `sql.errNoRows`. I think it should be fine. Although, the logic here is a bit complex and I do wish to simplify it.

The general flow is:
- Is there an article URL?
  - No: Do nothing, ignore rest of the steps
  - Yes: Check if it already exists
- Does article already exist?
  - No: Find the article data and add it to DB
  - Yes: Continue to next step
- Does the quiz already have the article?
  - No: Add it to the quiz
  - Yes: Continue to next step
- Add article to question and add question to DB.

Reply:
Actually, given that currently user cannot add new articles via "Create/edit question" window (that was not part of original plan, and changed recently), only articles that already exist in quiz, I can skip the "Does article already exist" and "Does the quiz already have the article" steps. Would that be preferable?